### PR TITLE
Reacting to new Hosting API

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Reacting to aspnet/Hosting#521

### CODE REVIEW

File: samples/DatabaseErrorPageSample/project.json [hunk 1]
```diff
@@ -4,12 +4,13 @@
...
-    "Microsoft.AspNet.Server.WebListener": "1.0.0-*",
     "Microsoft.AspNet.Server.Kestrel": "1.0.0-*"
   },
+  "compilationOptions": {
+    "emitEntryPoint": true
+  },
   "commands": {
-    "ef": "EntityFramework.Commands",
-    "web": "Microsoft.AspNet.Server.Kestrel"
+    "web": "DatabaseErrorPageSample"
```
Comment (JunTaoLuo) [inline samples/DatabaseErrorPageSample/project.json]: Removed the EF entrypoint here, do we need it?

File: samples/DeveloperExceptionPageSample/project.json [hunk 2]
```diff
@@ -2,10 +2,14 @@
   "webroot" : "wwwroot",
```
Comment (Tratcher) [inline samples/DeveloperExceptionPageSample/project.json]: Remove webroot from this file

File: samples/ElmPageSample/project.json [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
-    },
-    "frameworks": {
-        "dnx451" : { },
-        "dnxcore50" : { }
-    }
+  "webroot": "wwwroot",
+  "exclude": "wwwroot/**/*.*",
+  "dependencies": {
+    "Microsoft.AspNet.Diagnostics.Elm": "1.0.0-*",
+    "Microsoft.AspNet.Server.IIS": "1.0.0-*",
```
Comment (Tratcher) [inline samples/ElmPageSample/project.json]: Remove server.iis

File: samples/ExceptionHandlerSample/hosting.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+﻿{
+  "server": "Microsoft.AspNet.Server.Kestrel",
+  "server.urls": "http://localhost:5000"
```
Comment (Tratcher) [inline samples/ExceptionHandlerSample/hosting.json]: You remove server.urls if it's the default value.

**Comment by Tratcher** [comment]
:shipit:

**Comment by davidfowl** [comment]
I actually like this new API a lot better

**Comment by davidfowl** [comment]
:shipit:

File: samples/RuntimeInfoPageSample/Startup.cs [hunk 5]
```diff
@@ -20,5 +18,15 @@ public void Configure(IApplicationBuilder app)
                 return Task.FromResult(0);
             });
         }
+
+        public static void Main(string[] args)
+        {
+            var application = new WebApplicationBuilder()
+                .UseConfiguration(WebApplicationConfiguration.GetDefault(args))
```
Comment (davidfowl) [inline samples/RuntimeInfoPageSample/Startup.cs]: Is this too verbose?

Comment (Tratcher) [inline samples/RuntimeInfoPageSample/Startup.cs]: I asked elsewhere if we needed a `.UseConfiguration(args)` overload.

**Pull request merged**